evaluation/red_teaming.py

```diff
@@ -27,48 +30,70 @@ async def run_red_teaming(
     config: dict,
     red_teaming_llm: PromptChatTarget,
     prompt_target: PromptChatTarget,
+    max_turns: int,
+    compare: bool,
 ):
     """Run red teaming attack with provided scorers using Red Teaming Orchestrator."""
+    prompt_target_list = []
+    if compare and isinstance(prompt_target, AppChatTarget):
```

Comment:
Why are we dealing with the `AppChatTarget` separately, why does it matter what the target is? We could have multiple models on Azure, HF, or OpenAI, we might be evaluating the entire application or just the LLMs, it should all be the same from the red teaming POV.

Reply:
It matters because we do the comparison by changing the "set_model" param when requesting our app, and this param is set when setting up the AppChatTarget. Other LLM targets don't have this param when they are set up.

Reply:
> Why are we dealing with the `AppChatTarget` separately, why does it matter what the target is? We could have multiple models on Azure, HF, or OpenAI, we might be evaluating the entire application or just the LLMs, it should all be the same from the red teaming POV.

It matters because we do the comparison by changing the "set_model" param when requesting our app, and this param is set when setting up the AppChatTarget. Other LLM targets don't have this param when they are set up.

Reply:
ACK!

Reply:
> Why are we dealing with the `AppChatTarget` separately, why does it matter what the target is? We could have multiple models on Azure, HF, or OpenAI, we might be evaluating the entire application or just the LLMs, it should all be the same from the red teaming POV.

Do we need to implement red teaming comparison between LLMs? I think we can just compare the applications using different models, otherwise we need a totally different way to specify the model names rather than specifying in the config file.

Reply:
In all honesty, I've raised that before, the codebase should have been cleaned up, we don't care about supporting stuff on OpenAI or red teaming (or even evaluating) the models directly, it just creates clutter and adds no real value especially given that it's untested. I would leave everything as is for now.